mail_private/security/security.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+    <record id="message_private_rule" model="ir.rule">
+        <field name="model_id" ref="mail.model_mail_message"/>
+        <field name="name">Message private rules</field>
+        <field name="domain_force">['|', '|', ('mail_group_id', '=', False), ('model', '=', False), ('mail_group_id', 'in', user.message_group_ids.ids)]</field>
+    </record>
+
+
+    <record id="attachment_private_rule" model="ir.rule">
+        <field name="model_id" ref="base.model_ir_attachment"/>
+        <field name="name">Attachment private rules</field>
+        <field name="domain_force">['|', '|', '|', ('related_message_ids', '=', False),('related_message_ids.mail_group_id', '=', False), ('res_model', '=', False), ('related_message_ids.mail_group_id', 'in', user.message_group_ids.ids)]</field>
```

Comment:
After checking on a big database, this query is extremely inefficient. I will try to find another solution, although it does not seem easy...

Reply:
Well, I think I found a solution for this issue, setting the group of the attachment. Not as clean as I would like, but a solution after all